Source/Scene/ImageryLayer.js

```diff
@@ -969,6 +1149,197 @@ define([
+            persists : true,
+            owner : this,
+            // Update render resources right before execution instead of now.
+            // This allows different ImageryLayers to share the same vao and buffers.
+            preExecute : function(command) {
+                reprojectFromArbitrary(command, context, projectedTexture, outputTexture, rectangle, projectedRectangle, projection, positionsBuffer);
+            },
+            postExecute : function(outputTexture) {
+                if (final) {
+                    finalizeReprojectTexture(that, context, imagery, outputTexture);
+                    imagery.projectedTextures.length = 0;
+                }
+                imagery.releaseReference();
+            }
+        });
+    }
+
+    var geographicCartographicScratch = new Cartographic();
+    var projectedScratch = new Cartesian3();
+    function reprojectFromArbitrary(command, context, texture, outputTexture, cartographicRectangle, projectedRectangle, projection, positionsBuffer) {
+        var reproject = {
+            vertexArray : undefined,
+            shaderProgram : undefined,
+            sampler : undefined,
+            destroy : function() {
+                if (defined(this.framebuffer)) {
+                    this.framebuffer.destroy();
+                }
+                if (defined(this.vertexArray)) {
+                    this.vertexArray.destroy();
+                }
+                if (defined(this.shaderProgram)) {
+                    this.shaderProgram.destroy();
+                }
+            }
+        };
+
+        // For each vertex in the target grid, project into the projection
+        var south = cartographicRectangle.south;
+        var west = cartographicRectangle.west;
+
+        var unprojectedWidthIncrement = cartographicRectangle.width / (ARBITRARY_PROJECTION_VERTICES_WIDTH - 1);
+        var unprojectedHeightIncrement = cartographicRectangle.height / (ARBITRARY_PROJECTION_VERTICES_HEIGHT - 1);
+        var geographicCartographic = geographicCartographicScratch;
+        var projected = projectedScratch;
+
+        var projectedSouth = projectedRectangle.south;
+        var projectedWest = projectedRectangle.west;
+
+        var texcoords = new Float32Array(ARBITRARY_PROJECTION_VERTICES_WIDTH * ARBITRARY_PROJECTION_VERTICES_HEIGHT * 2);
+        var index = 0;
+        for (var w = 0; w < ARBITRARY_PROJECTION_VERTICES_WIDTH; w++) {
+            for (var h = 0; h < ARBITRARY_PROJECTION_VERTICES_HEIGHT; h++) {
+                geographicCartographic.longitude = west + w * unprojectedWidthIncrement;
+                geographicCartographic.latitude = south + h * unprojectedHeightIncrement;
+
+                projection.project(geographicCartographic, projected);
+
+                texcoords[index++] = (projected.x - projectedWest) / projectedRectangle.width;
+                texcoords[index++] = (projected.y - projectedSouth) / projectedRectangle.height;
+            }
+        }
```

Comment:
Why not generate all the projected coordinates up where the positions are generated and use a uniform in the vertex shader to map to `texcoords`? Possible precision issues?

Reply:
EPSG 3411 (polar stereographic) has bounds [-2353976.57, 2345773.95], [-382566.95, 383904.69], but I don't think you need to worry about the precision loss since you shouldn't be projecting an image over the entire area except when you're looking at it from very far away.